coverage-tool/coverage-tool.py

```diff
@@ -0,0 +1,300 @@
+# response that the files cover.
+#
+# Design doc: http://go/firebase-vertexai-test-coverage-tool-doc
+
+import json
+import os
+from argparse import ArgumentParser
+from glob import glob
+from urllib.request import urlopen
+
+DISCOVERY_DOC_URL = "https://aiplatform.googleapis.com/$discovery/rest?version=v1beta1"
+SCHEMA_PREFIX = "GoogleCloudAiplatformV1beta1"
+RESPONSE_TYPES = ["GenerateContentResponse", "CountTokensResponse"]
+SCRIPT_DIR = os.path.dirname(__file__)
+MOCK_RESPONSES_PATH = os.path.join(SCRIPT_DIR, "..", "mock-responses")
+
+NUM_KEYWORD = "_coverage_num"
+FILES_KEYWORD = "_coverage_files"
+TOTAL_KEYWORD = "_total_coverage_percentage"
+
+
+def get_args():
+    """Parse, validate, and return command line arguments."""
+
+    def get_files_from_patterns(pattern_list, root_dir="."):
+        files = set()
+        for pattern in pattern_list or []:
+            matches = {
+                path
+                for path in glob(pattern, recursive=True, root_dir=root_dir)
+                if os.path.isfile(os.path.join(root_dir, path))
+            }
+            if not matches:
+                parser.error(f"No matching files found for pattern: {pattern}")
+            files.update(matches)
+        return files
+
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--percent-only",
+        "-p",
+        action="store_true",
+        help="Print only the total coverage percentage",
+    )
+    parser.add_argument(
+        "--json-output",
+        "-j",
+        action="store_true",
+        help="Output the coverage data as JSON",
+    )
+    parser.add_argument(
+        "--list-files",
+        "-l",
+        action="store_true",
+        help="Output list of mock response files that cover each field",
+    )
+    parser.add_argument(
+        "--no-color",
+        "-n",
+        action="store_true",
+        help="Disable color in output",
+    )
+    parser.add_argument(
+        "--ignore-fields",
+        "-i",
+        metavar="FIELDS",
+        type=str,
+        help="Ignore the fields in the given comma-separated list when calculating"
+        " coverage",
+    )
+    parser.add_argument(
+        "--scan-files",
+        "-s",
+        metavar="PATTERN",
+        type=str,
+        nargs="+",
+        help="Scan files matching the given pattern(s) for names of mock response files"
+        " (without the extension) to evaluate",
+    )
+    parser.add_argument(
+        "--exclude",
+        "-e",
+        metavar="PATTERN",
+        type=str,
+        nargs="+",
+        help="Exclude mock response files matching the given pattern(s) from being"
+        " evaluated",
+    )
+    args = parser.parse_args()
+    if args.percent_only and (args.json_output or args.list_files):
+        parser.error(
+            "`--percent-only` cannot be used with `--json-output` or `--list-files`."
+        )
+    args.ignore_fields = (
```

Comment:
You can use `type` to do arg conversions

https://docs.python.org/3/library/argparse.html#type